hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/mint/MintTranslator.java

```diff
@@ -54,8 +57,27 @@ public boolean matches(@NonNull final HtsCallAttempt attempt) {
 
     @Override
     public HtsCall callFrom(@NonNull final HtsCallAttempt attempt) {
+        final var body = bodyForClassic(attempt);
+        final var isFungibleMint = body.tokenMintOrThrow().metadata().isEmpty();
```

Comment:
It's one thing to encode semantics (e.g., the "subtype" of something) the presence-or-absence of a field as long as it stays internal to the class.  Why can't we have a proper predicate on a `TokenMintTransactionBody` that exposes fungible or non-fungible? 

I know why, it's a protobuf, and for some reason (probably lack of time) we're stuck with having _only_ the API that Google's protobufs give us.  But protobufs are a communication protocol message, not an object-oriented domain type, and it's unfortunate the inability to specify domain-specific knowledge (predicates, in this case) is leaking through to us out of the communication layer.  (In our case it is also a storage layer, but same difference.)

Can this test be extracted into a nearby method so at least _here_ it looks like a proper test against a subtype?

(Remember all those questions we had earlier this year about how to distinguish fungible and non-fungible given the _amount_ being positive, zero, or negative, and the trouble that caused?  That's what I'm thinking of.)